ncm-network/src/main/perl/network.pm

```diff
@@ -1,1183 +1,1728 @@
-                $text = "DEVICE=$tmpdev\n";
-                if ($net{$iface}{aliases}{$al}{'ip'}) {
-                    $text .= "IPADDR=".$net{$iface}{aliases}{$al}{'ip'}."\n";
-                }
-                if ($net{$iface}{aliases}{$al}{'broadcast'}) {
-                    $text .= "BROADCAST=".$net{$iface}{aliases}{$al}{'broadcast'}."\n";
-                }
-                if ($net{$iface}{aliases}{$al}{'netmask'}) {
-                    $text .= "NETMASK=".$net{$iface}{aliases}{$al}{'netmask'}."\n";
-                }
-                $exifiles{$file_name} = $self->file_dump($file_name, $text, FAILED_SUFFIX);
-                $self->debug(3, "exifiles $file_name has value ".$exifiles{$file_name});
-            }
-        }
+    return \@text;
+}
+
+# Return ifcfg rule content in ip format
+# Very simple atm, only command supported.
+sub make_ifcfg_ip_rule
+{
+    my ($self, $flavour, $device, $rules) = @_;
 
+    my @text;
+    foreach my $rule (@$rules) {
+        $self->error("Rule flavour $flavour device $device without command; skipping") if (!$rule->{command});
+        push(@text, $rule->{command});
     }
 
-    # /etc/sysconfig/network
-    # assuming that NETWORKING=yes
-    $path=$base_path;
-    $file_name = "/etc/sysconfig/network";
-    mk_bu($file_name);
-    $exifiles{$file_name}=-1;
-    $text = "";
-    $text .= "NETWORKING=yes\n";
-    # set hostname.
-    if ($config->elementExists($path."/realhostname")) {
-        $text .= "HOSTNAME=".$config->getValue($path."/realhostname")."\n";
-    } else {
-        $text .= "HOSTNAME=".$config->getValue($path."/hostname").".".$config->getValue($path."/domainname")."\n";
+    return \@text;
+}
+
+
+# Return ifcfg alias content
+sub make_ifcfg_alias
+{
+    my ($self, $device, $alias) = @_;
+
+    my @text = ("DEVICE=$device");
+
+    my $makeline = _make_make_ifcfg_line($alias, \@text);
+    foreach my $attr (qw(ip broadcast netmask)) {
+        &$makeline($attr, var => ($attr eq 'ip') ? 'ipaddr' : undef);
     }
+
+    return \@text;
+}
+
+# Return /etc/sysconfig/network content
+sub make_network_cfg
+{
+    my ($self, $nwtree, $net, $hostname) = @_;
+
+    # assuming that NETWORKING=yes
+    my @text = ("NETWORKING=yes");
+
+    # set hostname
+    push(@text, "HOSTNAME=$hostname") if defined($hostname);
+
     # default gateway. why is this optional?
     #
     # what happens if no default_gateway is defined?
     # search for first defined gateway and use it.
     # here's the flag: default true
-    #
-    my $missing_default_gateway_autoguess = 1;
-    if ($config->elementExists($path."/guess_default_gateway")) {
-        if ($config->getValue($path."/guess_default_gateway") eq "false") {
-            $missing_default_gateway_autoguess = 0;
-        }
+    my $guess_dgw = defined($nwtree->{guess_default_gateway}) ? $nwtree->{guess_default_gateway} : 1;
+
+    my $nodgw_msg = "No default gateway configured";
+    my $dgw = $nwtree->{default_gateway};
```

Comment:
Is this ever used? It looks like it is simply overwritten if `$guess_dgw` is true.